### DISCUSSION TITLE
pressing any key make "BOOP" alert sound on macOS

### DISCUSSION THREAD

**Discussion by xingxing21** [AvaloniaUI/Avalonia#9381] [Q&A] [answered]
Hi
I'm hearing the "BOOP" alert sound when pressing any key that is not key-binding and is not processing in Key Down/Up event. 

1. Run an app
2. Click the top Window
3. Pressing Enter or Space or any key
4. make the "Boop" alert sound

- OS:  Mac
- Version 11.0.0-preview1

Thanks

**Comment by kekekeks** [comment, marked as answer]
You are supposed to mark the key event as handled, otherwise it's default Cocoa behavior to make a sound when you press some random keystroke.

**Reply by robloo** [reply]
Should this not be standardized by Avalonia across platforms?

**Reply by timunie** [reply]
I often agree with you @robloo , but in this particular field I am not sure. If a particular behavior is an OS standard, I think Avalonia should not break it. But because I don't own a Mac, I cannot finally judge.

**Reply by robloo** [reply]
At the very minimum it shouldn't require pulling in a behavior to fix for all TextBlocks. That said, other XAML frameworks like Uno don't do this and it seems quite unintuitive in certain scenarios. It is certainly open for discussion though as it's one of those things people could go either way on. However, the usefulness of cross-platform frameworks is standardization.

We might want to see what Maui and Flutter do as well.

Edit: added another comment for other frameworks. Seems they all boop...

**Reply by timunie** [reply]
I think you can handle this on any xaml tree level. If you want to handle it on Window stage the checks to perform may be different.

**Comment by xingxing21** [comment]
Yeah, you're right.
But if add the e.Handled = true; into the Key Down event on the top Window then can't type any key on a TextBox.
So if returning without e.Handled = true when TextBox has a focus and pressing the Enter Key on that TextBox then make a "BOOP" sound.
Also, if add the Key Down event into that TextBox with e.Handled = true then not but we should add the Key Down event into many TextBox.

**Comment by timunie** [comment]
You can use a [behavior](https://github.com/wieslawsoltes/AvaloniaBehaviors) for example which can be added via Style to all TextBoxes.

**Comment by timunie** [comment]
According to @kekekeks and your confirmation I think this is by design. So I convert this issue into a Q&A discussion. If you disagree, let me know.

**Reply by xingxing21** [reply]
Yes, I agree.
This is one solution but are you going to suggest all the developer this method?
Is there not a way?

**Reply by robloo** [reply]
Avalonia should try to standardize this across platforms. This type of thing is unintuitive and shouldn't require a behavior to fix. IMO the default in Avalonia should be to not play this sound on unhandled key presses. You should have to turn this on with a platform config.

**Comment by robloo** [comment]
Benchmarking:

Catalyst : https://developer.apple.com/forums/thread/651252

Flutter : https://github.com/flutter/flutter/issues/74287

Maui : Can't find explicitly and probably needs a quick test. Seems reasonable to assume it behaves the same as catalyst though.
